### PR TITLE
TRUNK-5134: Replace path constructions using separator char

### PR DESCRIPTION
**Pull request by brensurio**
## Description of what I changed

Modified 5 files for this task:

- api/src/main/java/org/openmrs/util/UpgradeUtil.java
- web/src/main/java/org/openmrs/module/web/WebModuleUtil.java
- web/src/main/java/org/openmrs/web/filter/initialization/TestInstallUtil.java
- web/src/test/java/org/openmrs/module/web/WebModuleUtilTest.java
- web/src/test/java/org/openmrs/web/test/WebModuleActivatorTest.java

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-5134

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: web/src/main/java/org/openmrs/module/web/WebModuleUtil.java [hunk 1]
```diff
@@ -770,7 +770,7 @@ public static void stopModule(Module mod, ServletContext servletContext, boolean
 			if (root.getElementsByTagName("dwr").getLength() > 0) {
 				
 				// get the dwr-module.xml file that we're appending our code to
-				File f = new File(realPath + "/WEB-INF/dwr-modules.xml".replace("/", File.separator));
+				File f = new File(realPath, "/WEB-INF/dwr-modules.xml");
```
Comment (wluyima) [inline web/src/main/java/org/openmrs/module/web/WebModuleUtil.java]: I thought you need to get rid of the forward slashes, this is where Paths comes in handy

File: web/src/main/java/org/openmrs/module/web/WebModuleUtil.java [hunk 2]
```diff
@@ -948,8 +948,7 @@ public static void createDwrModulesXml(String realPath) {
 			TransformerFactory transformerFactory = TransformerFactory.newInstance();
 			Transformer transformer = transformerFactory.newTransformer();
 			DOMSource source = new DOMSource(doc);
-			StreamResult result = new StreamResult(new File(realPath
-			        + "/WEB-INF/dwr-modules.xml".replace("/", File.separator)));
+			StreamResult result = new StreamResult(new File(realPath, "/WEB-INF/dwr-modules.xml"));
```
Comment (wluyima) [inline web/src/main/java/org/openmrs/module/web/WebModuleUtil.java]: You need to get rid of the forward slashes here too

File: web/src/main/java/org/openmrs/web/filter/initialization/TestInstallUtil.java [hunk 3]
```diff
@@ -68,8 +68,7 @@ protected static boolean addTestData(String host, int port, String databaseName,
 		
 		//For stand-alone, use explicit path to the mysql executable.
 		String runDirectory = System.getProperties().getProperty("user.dir");
-		File file = new File(runDirectory + File.separatorChar + "database" + File.separatorChar + "bin"
-		        + File.separatorChar + "mysql");
+		File file = new File(runDirectory, "/database/bin/mysql");
```
Comment (wluyima) [inline web/src/main/java/org/openmrs/web/filter/initialization/TestInstallUtil.java]: You need to get rid of the forward slashes here too

File: web/src/test/java/org/openmrs/module/web/WebModuleUtilTest.java [hunk 4]
```diff
@@ -112,7 +112,7 @@ public void startModule_shouldCreateDwrModulesXmlIfNotExists() throws ParserConf
 			realPath = System.getProperty("user.dir");
 		
 		// manually delete dwr-modules.xml 
-		File f = new File(realPath + "/WEB-INF/dwr-modules.xml");
+		File f = new File(realPath, "/WEB-INF/dwr-modules.xml");
```
Comment (wluyima) [inline web/src/test/java/org/openmrs/module/web/WebModuleUtilTest.java]: You need to get rid of the forward slashes here too

File: web/src/test/java/org/openmrs/web/test/WebModuleActivatorTest.java [hunk 5]
```diff
@@ -41,8 +41,8 @@ public void createWebInfFolderIfNotExist() {
 		//when run from the IDE and this folder does not exist, some tests fail with
 		//org.openmrs.module.ModuleException: Unable to load module messages from file: 
 		// /Projects/openmrs/core/web/target/test-classes/WEB-INF/module_messages_fr.properties
-		
-		File folder = new File("target" + File.separatorChar + "test-classes" + File.separatorChar + "WEB-INF");
+
+		File folder = new File("target/test-classes/WEB-INF");
```
Comment (wluyima) [inline web/src/test/java/org/openmrs/web/test/WebModuleActivatorTest.java]: You need to get rid of the forward slashes here too

File: web/src/main/java/org/openmrs/module/web/WebModuleUtil.java [hunk 6]
```diff
@@ -237,7 +237,7 @@ public static boolean startModule(Module mod, ServletContext servletContext, boo
 				if (root.getElementsByTagName("dwr").getLength() > 0) {
 					
 					// get the dwr-module.xml file that we're appending our code to
-					File f = new File(realPath + "/WEB-INF/dwr-modules.xml".replace("/", File.separator));
+					File f = new File(realPath, "/WEB-INF/dwr-modules.xml");
```
Comment (wluyima) [inline web/src/main/java/org/openmrs/module/web/WebModuleUtil.java]: You need to get rid of the forward slashes here too

**Comment by brensurio** [comment]
Thanks, I'll continue working on this.

**Comment by brensurio** [comment]
I forgot to squash. I'm working on it.

**Comment by brensurio** [comment]
I removed the slashes mentioned in review comment. For that, I used `Paths.get()` then converted it to string in order to still use `File` constructor as specified from the ticket.

File: web/src/main/java/org/openmrs/module/web/WebModuleUtil.java [hunk 7]
```diff
@@ -237,7 +238,7 @@ public static boolean startModule(Module mod, ServletContext servletContext, boo
 				if (root.getElementsByTagName("dwr").getLength() > 0) {
 					
 					// get the dwr-module.xml file that we're appending our code to
-					File f = new File(realPath + "/WEB-INF/dwr-modules.xml".replace("/", File.separator));
+					File f = new File(Paths.get(realPath, "WEB-INF", "dwr-modules.xml").toString());
```
Comment (wluyima) [inline web/src/main/java/org/openmrs/module/web/WebModuleUtil.java]: Paths has a `toFile()` Method , please use that

File: web/src/main/java/org/openmrs/module/web/WebModuleUtil.java [hunk 8]
```diff
@@ -286,7 +287,7 @@ public static boolean startModule(Module mod, ServletContext servletContext, boo
 			// mark to delete the entire module web directory on exit
 			// this will usually only be used when an improper shutdown has occurred.
 			String folderPath = realPath + "/WEB-INF/view/module/" + mod.getModuleIdAsPath();
```
Comment (wluyima) [inline web/src/main/java/org/openmrs/module/web/WebModuleUtil.java]: Note that folderPath on the line above contains forward slashes, you need to get rid of them too

File: web/src/main/java/org/openmrs/module/web/WebModuleUtil.java [hunk 9]
```diff
@@ -173,7 +174,7 @@ public static boolean startModule(Module mod, ServletContext servletContext, boo
 						}
 						
 						// get the output file
-						File outFile = new File(absPath.toString().replace("/", File.separator));
+						File outFile = new File(absPath.toString());
```
Comment (wluyima) [inline web/src/main/java/org/openmrs/module/web/WebModuleUtil.java]: Doesn't `absPath` contain a '/'? I think it does

File: web/src/main/java/org/openmrs/module/web/WebModuleUtil.java [hunk 10]
```diff
@@ -689,7 +690,7 @@ public static void shutdownModules(ServletContext servletContext) {
 		
 		// clear the module messages
 		String messagesPath = realPath + "/WEB-INF/";
-		File folder = new File(messagesPath.replace("/", File.separator));
+		File folder = new File(messagesPath);
```
Comment (wluyima) [inline web/src/main/java/org/openmrs/module/web/WebModuleUtil.java]: messagesPath contains a forward slash

**Comment by wluyima** [comment]
You need to get rid of forward slashes in all paths.

**Comment by dkayiwa** [comment]
@brensurio are you still working on this?

**Pull request closed without merging**